### ISSUE TITLE
Applying fix for regexp/optimal-quantifier-concatenation break the regex logic

### ISSUE THREAD

**Issue by Eugeno** [ota-meshi/eslint-plugin-regexp#736]
**Information:**
- ESLint version: 8.57.0
- `eslint-plugin-regexp` version: 2.5.0

**Description**
Current regex is `/^(?!.*(?=.*-)(?=.*,)).*$/` (disallow usage of both comma and dash).
```js
/^(?!.*(?=.*-)(?=.*,)).*$/.test('1,2-3'); // false ✓
```
I got 3 errors here, one of them: `'.*' can be removed because it is already included by '.*'.(regexp/optimal-quantifier-concatenation)`.
After applying the fix:
```js
/^(?!.*(?=-)(?=.*,)).*$/.test('1,2-3'); // true ✗
```
Further autofixes give the same false positive result:
```js
/^(?!.*(?=-).+,).*$/.test('1,2-3'); // true ✗
```

**Comment by RunDevelopment** [comment]
Very good fine. Thanks for reporting.

The transformation:

```js
/^(?!.*(?=.*-)(?=.*,)).*$/;
// to
/^(?!.*(?=-)(?=.*,)).*$/;
```

Is indeed incorrect. I think I messed one of its conditions, so this shouldn't be too hard to fix.
